### ISSUE TITLE
'init' keyword on Proj() causing segfault

### ISSUE THREAD

**Issue by Thingus** [pyproj4/pyproj#367]
```python
import pyproj
pyproj.Proj('EPSG:4326', preserve_units=True)  # Works
pyproj.Proj(init='EPSG:4326', preserve_units=True) # Throws segfault
```
#### Problem description
When a new Proj is instantitated, creating with the ```init=``` keyword arg throws a segfault.

#### Expected output
Either work as expected or throw kwargs exception? This feels related to https://pyproj4.github.io/pyproj/stable/gotchas.html
If this is a known wont-fix, please remove this report.

#### Environment Information
System:
    python: 3.7.0 (default, Jun 28 2018, 13:15:42)  [GCC 7.2.0]
executable: /home/ubuntu/tools/miniconda/bin/python
   machine: Linux-4.15.0-1023-aws-x86_64-with-debian-buster-sid

PROJ:
      PROJ: 6.1.0
  data dir: /home/ubuntu/tools/miniconda/lib/python3.7/site-packages/pyproj/proj_dir/share/proj

Python deps:
    pyproj: 2.2.1
       pip: 19.1.1
setuptools: 40.2.0
    Cython: None
     aenum: None

#### Installation method
Installed from pip

#### Conda environment information (if you installed with conda):

<br/>
Environment (<code>conda list</code>):
<details>

```
proj4                     5.0.1                h14c3975_0  
pyproj                    2.2.0                     <pip>


```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

     active environment : pyeo_env
    active env location : /home/ubuntu/tools/miniconda/envs/pyeo_env
            shell level : 1
       user config file : /home/ubuntu/.condarc
 populated config files : 
          conda version : 4.5.11
    conda-build version : not installed
         python version : 3.7.0.final.0
       base environment : /home/ubuntu/tools/miniconda  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/linux-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/pro/linux-64
                          https://repo.anaconda.com/pkgs/pro/noarch
          package cache : /home/ubuntu/tools/miniconda/pkgs
                          /home/ubuntu/.conda/pkgs
       envs directories : /home/ubuntu/tools/miniconda/envs
                          /home/ubuntu/.conda/envs
               platform : linux-64
             user-agent : conda/4.5.11 requests/2.19.1 CPython/3.7.0 Linux/4.15.0-1023-aws ubuntu/18.04 glibc/2.27
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False

</details>

**Comment by jorisvandenbossche** [comment]
You mention you have PROJ 6.1, but the `conda list` output shows proj4 5.0.1. So those probably conflict, giving the segfault.

It seems you installed pyproj with pip. Can you try to remove it (`pip uninstall pyproj`) and then install it with conda and see if this works?

**Comment by snowman2** [comment]
Mind trying out: `pip install http://a365fff413fe338398b6-1c8a9b3114517dc5fe17b7c3f8c63a43.r19.cf2.rackcdn.com/pyproj-2.2.2rc0-cp36-cp36m-manylinux1_x86_64.whl`?

**Comment by jorisvandenbossche** [comment]
@snowman2 Out of curiosity, what changed i the recent wheels that might have fixed this overlap with another proj install?

**Comment by snowman2** [comment]
Probably nothing. I made this change #361 and it is probably irrelevant, but I am curious.

**Comment by snowman2** [comment]
It didn't change anything. Still dumped.

**Comment by snowman2** [comment]
Seems like an issue is needed for looking into the mix of the PROJ versions.

**Comment by Thingus** [comment]
> The solution by @jorisvandenbossche should resolve this. But, for my curiosity sake, mind trying out: `pip install http://a365fff413fe338398b6-1c8a9b3114517dc5fe17b7c3f8c63a43.r19.cf2.rackcdn.com/pyproj-2.2.2rc0-cp36-cp36m-manylinux1_x86_64.whl`?

Tried this, but it's not a valid wheel for my platform.

It looks like it might be the fault of my untidy environment - I'll try @jorisvandenbossche 's solution, and then clean this up and start from scratch.

**Comment by Thingus** [comment]
> You mention you have PROJ 6.1, but the `conda list` output shows proj4 5.0.1. So those probably conflict, giving the segfault.
> 
> It seems you installed pyproj with pip. Can you try to remove it (`pip uninstall pyproj`) and then install it with conda and see if this works?

This resolved it - thanks.
Pyproj was installed as part of a different package - it might be a mismatch from there. I'll go look at their specification and see if it's the same problem.

**Comment by snowman2** [comment]
Closing for preference to #368

**Comment by snowman2** [comment]
Fix in soon to be released 2.3.1
